lib/services/local/assistant_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+ *
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * /
+ */
+
+package local_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/client/proto"
+)
+
+// TestAssistantCRUD tests the assistant CRUD operations.
+func TestAssistantCRUD(t *testing.T) {
+	t.Parallel()
+
+	identity := newIdentityService(t, clockwork.NewFakeClock())
+	ctx := context.Background()
+
+	const username = "foo"
+	var conversationID string
+
+	t.Run("create conversation", func(t *testing.T) {
+		req := &proto.CreateAssistantConversationRequest{
+			Username:    username,
+			CreatedTime: time.Now(),
+		}
+
+		conversationResp, err := identity.CreateAssistantConversation(ctx, req)
+		require.NoError(t, err)
+		require.NotEmpty(t, conversationResp.Id)
+
+		conversationID = conversationResp.Id
+	})
+
+	t.Run("get conversation", func(t *testing.T) {
+		req := &proto.GetAssistantConversationsRequest{
+			Username: username,
+		}
+		conversations, err := identity.GetAssistantConversations(ctx, req)
+		require.NoError(t, err)
+		require.Len(t, conversations.Conversations, 1)
+		require.Equal(t, conversationID, conversations.Conversations[0].Id)
+	})
+
+	t.Run("create message", func(t *testing.T) {
+		msg := &proto.AssistantMessage{
+			Username:       username,
+			CreatedTime:    time.Now(),
+			ConversationId: conversationID,
+			Payload:        "foo",
+			Type:           "USER_MSG",
```

Comment:
Do we have a constant for `USER_MSG` somewhere? Arbitrary hard-coded strings when there are only a few valid types seems like a bad idea.

Reply:
This is a test, and in this case, the value doesn't make a huge difference. I can remove it if you think it's confusing.